Add upsell nudge for sites that have AMP deprecated.

wojtekn opened the pull request:
#### Proposed Changes

We are in the process of removing AMP for Simple sites. As the option that disappeared from the settings may be confusing, we are adding a notice that displays a message asking to install the AMP plugin.

![amp-nudge](https://user-images.githubusercontent.com/727413/176371227-2c6ef84e-e2c5-47a7-a448-9fc820448763.png)

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

##### Preconditions
* ~~use sandbox with D83077-code applied (if it's not deployed yet)~~

##### Simple site with AMP deprecated, without a paid plan

1. Navigate to "WP Admin -> Settings -> Performance"
2. Confirm that the AMP box displays a message that asks to install the AMP plugin
3. Click on the message
4. Confirm that the AMP plugin is shown and CTA says "Upgrade and activate"

##### Simple site with AMP deprecated, with a paid plan

1. Navigate to "WP Admin -> Settings -> Performance"
2. Confirm that the AMP box displays a message that asks to install the AMP plugin
3. Click on the message
4. Confirm that the AMP plugin is shown and CTA says "Install and activate"

##### Simple site that is private

1. Navigate to "WP Admin -> Settings -> Performance"
2. Confirm that the AMP box is not displayed at all

##### Simple site with AMP not deprecated

1. Navigate to "WP Admin -> Settings -> Performance"
2. Confirm that the AMP settings are still there

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/64731

wojtekn committed:
Add upsell nudge for sites that have AMP deprecated.

I'm adding an upsell nudge that displays two versions of the nudge:
- for a user with a free account
- for a user who has paid plan but has not enabled hosting features yet.

sejas reviewed (approved):
I've tested the different cases and everything works great.
- [x] Simple site with AMP deprecated, without a paid plan
- [x] Simple site with AMP deprecated, with a paid plan
- [x] Simple site that is private
- [x] Simple site with AMP not deprecated

The  only weird thing I've seen is that the `Jetpack CDN` and `Cloudflare CDN` have the same CTA `Available on the Pro plan` but they link to the the plans page, and we are linking to the checkout page. I understand the reason why we need a pro plan (we need to install a plugin).
But I'm wondering if those plans should like to the checkout page as well. And maybe we need to create an issue to change that link.

<img width="723" alt="Screenshot 2022-06-23 at 16 04 09" src="https://user-images.githubusercontent.com/779993/175331236-a14dba6a-0736-4ab4-963c-a2172d04a504.png">

wojtekn committed:
Configure upsell nudge impression and click events

We define impression and click events as UpsellNudge attributes
so we can utilize the OOTB support for tracking those.

wojtekn committed:
Remove the upgrade plan nudge as it is handled by plugin installation

Message that asks user to upgrade to the Pro plan in order to
install the AMP plugin is not needed, as the plan upgrade is
already handled by plugin installation flow.

wojtekn commented:
Thanks, @sejas. I've simplified the logic as we discovered that we don't need two versions of the upsell there.

I've also added another ticket with the CDN call to action issue.

sejas reviewed (approved):
@wojtekn Everything is working great. I like the new approach to link to the AMP plugin page. I think it's a better and smarter solution. Well done!

# Simple Sites (Private/Coming Soon)
Doesn't show any AMP information.

# Simple Sites (Public)
Links to the plugin.

**Screenshot**
<img width="752" alt="Screenshot 2022-06-27 at 21 22 01" src="https://user-images.githubusercontent.com/779993/176028851-b55bfd36-c728-451f-86fb-2c7488efae05.png">

**Screencast**

https://user-images.githubusercontent.com/779993/176029363-0998acc9-d5cc-4d04-bffd-3b609cd04c67.mp4

# Simple Non Deprecated

<img width="778" alt="Screenshot 2022-06-27 at 21 36 51" src="https://user-images.githubusercontent.com/779993/176031243-f0326b89-74ea-4a06-ad98-c7946ac737cf.png">


# Atomic
I've also tested an atomic site using calypso.
We are showing the [AmpJetpack component](https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/my-sites/site-settings/amp/jetpack.jsx#L13), and not any other box.
<img width="747" alt="Screenshot 2022-06-27 at 21 19 41" src="https://user-images.githubusercontent.com/779993/176028499-db6c4826-d4f6-4888-9b3d-f1a9bb48f01c.png">

p-jackson reviewed (commented):
Looking good. Great to see the upsell component already supports a standard way of doing tracks 👍 

> Install AMP plugin.

I think we should drop the trailing full-stop from this title since the other upsells on this page don't have it.

> Install AMP plugin to use AMP features.

This message feels a little too austere IMO.

We could copy just copy the same message from the Jetpack UI: `AMP enables the creation of websites and ads that load near instantly, giving site visitors a smooth,
more engaging experience on mobile and desktop.`

I like that it mentions mobile _and_ desktop, which is a new thing you get when you're not running the plugin in legacy mode. I also like that if we copy it then it'll already be translated :)

wojtekn committed:
Make AMP message wording gentler

wojtekn committed:
Reuse existing phrase for AMP installation nudge

Add missing "the" article and reuse text that is
already used in codebase and which is already translated.

wojtekn commented:
Thanks, @sejas, and @p-jackson.

Finally, I've used the following wording as I found out that this one is also already used in the codebase so that we can benefit from the existing translation too:

> Install the AMP plugin

wojtekn commented:
I've updated the ticket description to include changes we made during the review and deployed those.
